### PR TITLE
Guide

### PR DESCRIPTION
**Pull request by sornas**

### CODE REVIEW

File: docs/guide.adoc [hunk 1]
```diff
@@ -1 +1,73 @@
...
+"sometimes \"escaped quotes\" as well"
+
+// Integers
+1
+-2
+
+// Floats
+1.0
+-5.5
+1.    // trailing 0 optional
```
Comment (FredTheDino) [inline docs/guide.adoc:66]: .1 aswell?

**Review by FredTheDino — COMMENTED** [review]
Cool start!

File: docs/guide.adoc [hunk 2]
```diff
@@ -1 +1,174 @@
...
+----
+true or false
+true and true
+not true
+----
+
+== Imports
+
+Code can be written in multiple files, to your liking. You don't have to
+consider include ordering and dependency cycles. Write your code anywhere you
```
Comment (EMattfolk) [inline docs/guide.adoc:152]:
```suggestion
consider include ordering or dependency cycles. Write your code anywhere you
```

File: docs/guide.adoc [hunk 3]
```diff
@@ -1 +1,174 @@
...
+false
+----
+
+=== Composite values
+
+== If-statements and if-expressions
+
+These are your normal control flow statements.
+
+.Normal if-statements
```
Comment (EMattfolk) [inline docs/guide.adoc:80]: I would change "normal" to "standard" or "regular".

File: docs/guide.adoc [hunk 4]
```diff
@@ -1 +1,174 @@
...
+----
+// These two are equivalent
+a := (b + 5.) / 10. if (b + 5.) / 10. > 1. else 0.
+//   ^^^^^^^^^^^^^^    ^^^^^^^^^^^^^^
+a := (b + 5.) / 10. if > 1. else 0.
+----
+
+== Operators
+
+The basic operators are all here. Normal mathematical grouping is applied.
```
Comment (EMattfolk) [inline docs/guide.adoc:110]:
```suggestion
The basic operators are all here. Standard mathematical evaluation order applies.
```
Or something simiar. This line might need a linebreak.

**Comment by sornas** [comment]
I added a couple of headers that can be filled in. I want to look over the ordering of the headers before merging.

File: docs/guide.adoc [hunk 5]
```diff
@@ -1,2 +1,262 @@
...
+.Functions don't have to be constructed in-place.
+[source, sylt]
+----
+Player :: blob {
+    pos: (int, int)
+    vel: (int, int)
+}
+
+update_player :: fn p: Player do
+    pos += vel
```
Comment (EMattfolk) [inline docs/guide.adoc:251]:
```suggestion
    p.pos += p.vel
```

File: docs/guide.adoc [hunk 6]
```diff
@@ -1,2 +1,262 @@
...
+couple of higher-order functions. Still, they are written almost like a the
+normal loops
+
+=== for_each
+
+[source, sylt]
+----
+l := [1, 2, 3]
+
+l -> for_each' fn a: int do print a end
```
Comment (EMattfolk) [inline docs/guide.adoc:235]:
```suggestion
l -> for_each' fn a: int do print(a) end
```

**Pull request merged**